src/plugins/data/public/ui/dataset_selector/configurator.tsx

```diff
@@ -51,6 +52,17 @@ export const Configurator = ({
     return languages[0];
   });
 
+  const displayTimeFieldSelector = (ds: Dataset, lang: string, fields: DatasetField[]) => {
+    return (
+      !(
+        languageService.getLanguage(lang)?.disableDatePicker ||
+        ds.type === DEFAULT_DATA.SET_TYPES.INDEX_PATTERN
+      ) &&
+      fields &&
+      fields.length > 0
+    );
+  };
+
   useEffect(() => {
     const fetchFields = async () => {
       const datasetFields = await queryString
```

Comment:
we should add a follow up or a spinner to prevent the date field from loading in afterwards